ToolBox/MainThreadDispatcher.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MainThreadDispatcher : MonoBehaviour
```

Comment:
You can use ThreadingHelper from BepInEx instead to the same effect (`ThreadingHelper.Instance.StartSyncInvoke`), no need to make a new class for it.